cloudevents/working-drafts/xml-format.md

```diff
@@ -0,0 +1,277 @@
+| URI           | [xs:anyURI][xml-primitives] following [RFC 3986][rfc3986]| |
+| URI-reference | [xs:anyURI][xml-primitives] following [RFC 3986][rfc3986] | |
+| Timestamp     | [xs:dateTime][xml-primitives] following [RFC 3339][rfc3339] | |
+
+Each context attribute is represented as an XML element whose name MUST exactly match
+that of a REQUIRED or OPTIONAL CloudEvent attribute, extension attribute names MUST
+adhere to, and align with, the conventions in defined in the [cloud event specification][ce-spec].
+
+See the [envelope](#4-envelope) for special handing of the `specversion` context attribute.
+
+Extension attributes SHOULD be expressed with the appropriate `xsi:type` to allow them
+to be exchanged without loss of type information. Extension attributes with no `xsi:type`
+discriminator SHOULD be interpreted with an implied type of `xs:string`
+
+``` xml
+    ...
+    <id>AAABBBCCCNNN0000</id>
+    ..
+    <time>2021-08-14T14:30:22-08:00</time>
+    ..
+    <myextension xsi:type="xs:string">my extension value</myextension>
+    <myboolean xsi:type="xs:boolean">false</myboolean>
+```
+
+## 3. Data
+
+The `Data` portion of a CloudEvent follows a similar model to that employed by
+the [JSON Format specification][json-format]. A `data` element is used to
+encapsulate the payload and an `xsi:type` used to discrimate the payload
+type.
+
+### 3.1 Binary Data
+
+MUST be carried in an element with an defined type of `xs:base64Binary`
+
+``` xml
+<data xsi:type="xs:base64Binary">.........</data>
+```
+
+### 3.2 Text Data
+
+MUST be carried in an element with an defined type of `xs:string`
+
+``` xml
+<data xsi:type="xs:string">This is text</data>
+```
+
+### 3.3 XML Data
+
+XML data MUST be carried in an element with a defined type of `xs:any`
+allowing a single XML element (with any required namespace definitions)
+to be represented.
+
+``` xml
+<data xsi:type="xs:any">
+    <myData>
+        ....
+    </myData>  
+</data>
+```
+
+## 4. Envelope
+
+Each CloudEvent is wholly represented as an XML element `<Event>` that
+carries the `specversion` as an XML attribute value.
+
+Such a representation MUST use the media type `application/cloudevents+xml`.
+
+The enveloping element contains:
+
+* A set of context attribute elements.
+* An optional `data` element.
+
+eg _(XML preamble and namespace definitions omitted for brevity)_:
+
+``` xml
+<Event specversion="1.0">
+    <time>2020-03-19T12:54:00-07:00</time>
+    <datacontenttype>text/plain</datacontenttype>
+    <id>000-1111-2222</id>
+    <source>urn:uuid:123e4567-e89b-12d3-a456-426614174000</source>
+    <type>SOME.EVENT.TYPE</type>
+    <myboolean xsi:type="xs:boolean">false</myboolean>
+    <data xsi:type="xs:string">Now is the winter of our discount tents...</data>
+</Event>
+```
+
+## 5. XML Batch Format
+
+In the _XML Batch Format_ several CloudEvents are batched into a single XML
+document. The document comprises a list of elements in the XML Format.
```

Comment:
document => element (unless this is really trying to say that you shouldn't create a document that includes multiple batches)

Reply:
ack